File: src/java/jnpp/service/dto/accounts/ShareAccountDTO.java
package jnpp.service.dto.accounts;

import java.util.ArrayList;
import java.util.List;

public class ShareAccountDTO extends AccountDTO {

    private List<ShareTitleDTO> shareTitles;

    public ShareAccountDTO(String rib, List<ShareTitleDTO> shareTitles) {
        super(rib);
        this.shareTitles = new ArrayList<ShareTitleDTO>(shareTitles);
    }

    @Override
    public Type getType() {
        return AccountDTO.Type.SHARE;
    }

    public List<ShareTitleDTO> getShareTitles() {
        return shareTitles;
    }

    public void setShareTitles(List<ShareTitleDTO> shareTitles) {
        this.shareTitles = shareTitles;
    }

}

File: src/main/webapp/commons/dto/accounts/share-account-dto.js
class ShareAccountDTO extends AccountDTO {

    constructor(json) {
        super(json, AccountDTO.Type.SHARE);
        if (!json.shareTitles) throw new ParseException(json, typeof this);
        this.shareTitles = new Array();
        for (let i = 0; i < json.shareTitles.length; ++i)
            this.shareTitles.push(new ShareTitleDTO(json.shareTitles[i]));
    }

}
